### PR TITLE
[Cache] Add tags based invalidation + TagAwareRedisAdapter

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #18646 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Rebased, RFR

File: src/Symfony/Component/Cache/README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Symfony PSR-6 implementation for caching
 ========================================
 
-This component provides a strict [PSR-6](http://www.php-fig.org/psr/psr-6/)
+This component provides a extended [PSR-6](http://www.php-fig.org/psr/psr-6/)
```
Comment (fabpot) [inline src/Symfony/Component/Cache/README.md]: an extended

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by aik099** [comment]
@nicolas-grekas ,

1. I see, that `TagAwareRedisAdapter` was added here, but I don't see it anymore in Symfony 3.3.9. Was it removed?
2. How do I create taggable cache adapter from file (for tag storage) and memcached (for data storage) adapters and then set it via `framework.cache.app` config option in `app/config/config.yml`?

Thanks.

**Comment by xabbuh** [comment]
@aik099 The `TagAwareRedisAdapter` was replaced by a generic `TagAwareAdapter` in #19524.

**Comment by nicolas-grekas** [comment]
@aik099 `TagAwareRedisAdapter` has never been released. Use `TagAwareAdapter` instead. For you other question, I invite you tu use a support channel (we use github for dev, not support , see http://symfony.com/support)

**Comment by aik099** [comment]
I've tried `#support` channel on Slack, but apparently nobody in there is using `TagAwareAdapter` and therefore we weren't able to get actually working solution :(

**Pull request merged**